### ISSUE TITLE
mpv is a bad example

### ISSUE THREAD

**Issue by vitorgalvao** [cdown/yturl#25]
Installing `yturl` and doing  `mpv "$(yturl 'http://www.youtube.com/watch?v=8TCxE0bWQeQ')"` doesn’t make sense. If you install [`youtube-dl`](https://rg3.github.io/youtube-dl/) (which has an even more open license than `yturl`, since it is public domain), you can just `mpv 'http://www.youtube.com/watch?v=8TCxE0bWQeQ'`.

When `youtube-dl` is installed, `mpv` automatically takes advantage of it, so it is not at all the best example to showcase what `yturl` can do, since you’re actually making the command worse.

**Comment by cdown** [comment]
There's a reason people don't use youtube-dl -- for one, because it's utterly humongous. But I won't go into design and culture differences here, there's no point.

I have not much idea about why the license matters. The only reason I license as MIT is because it's a license that everyone is familiar with -- a license which people feel comfortable contributing with. Many people don't feel that way with public domain code (regardless of "correctness" for their feelings).

I think saying that it "makes the command worse" is quite a strange thing to say, since mpv can't do this natively. youtube-dl is a dependency for that action, and there's a reason people choose different software.

The point of yturl was always to be a minimal client for viewing YouTube videos. youtube-dl exists far out of that scope, and has done for a long time.

**Comment by cdown** [comment]
Continuing discussion of comparison with youtube-dl in #24, but the rest is not really worth discussing (since they are not actionable questions).

**Comment by cdown** [comment]
Whoops, didn't mean to do that.

**Comment by cdown** [comment]
As for public domain licensing, I was under the errant impression that this wasn't possible to do from a legal perspective. Seems that's not true: http://cr.yp.to/publicdomain.html

**Comment by cdown** [comment]
Here are some concerns about Unlicense itself that seem valid: https://lists.opensource.org/pipermail/license-review/2012-January/000052.html

**Comment by vitorgalvao** [comment]
Thank you for the fast and polite replies. I’ll try to address all of your points.

> There's a reason some people prefer not to use youtube-dl -- for one, because it's utterly humongous.

Oh, I agree there. To be perfectly clear, I never once said (or even meant to imply) `youtube-dl` is superior in every way and there is no place for `yturl`. My point concerned that sole command and only because, as I see it, *it makes `yturl` seem worse in comparison*. If you see a command and think “wait, I can do it easier the other way” you lose that person. The sole reason I mentioned `youtube-dl` specifically is it is the package `mpv` uses, nothing more.

> But I won't go into design and culture differences here, there's no point.

Very much agreed.

> I think saying that it "makes the command worse" is quite a strange thing to say, since mpv can't do this natively.

But it also doesn’t do it natively with `yturl`. Think of it this way: you need to install one of two external command (`youtube-dl` or `yturl`) to get this capability. One of them makes the command more verbose and less straightforward (having to run a command as an argument, as opposed to simply giving the desired URL). Seeing as the result is the same to the user, the simpler command is better.

> The point of yturl was always to be a minimal client for viewing YouTube videos. youtube-dl exists far out of that scope, and has done for a long time.

That line alone could close https://github.com/cdown/yturl/issues/24 (great reply there, as well).

As for licensing stuff:

> As for public domain licensing, I was under the errant impression that this wasn't possible to do from a legal perspective. Seems that's not true: http://cr.yp.to/publicdomain.html
> (…)
> Here are some concerns about Unlicense itself that seem valid: https://lists.opensource.org/pipermail/license-review/2012-January/000052.html

Thank you for the links. Will be sure to read them. Only skimmed yet, and was already corrected on the German situation (I’m not German, but still was under that impression).

On a first pass, though, I’ll say The Unlicense is not the only popular Public Domain license (there’s also [CC0](https://creativecommons.org/publicdomain/zero/1.0/), and that one is more full of legalese and itself tries to address concerns with other licenses) but it is itself provided as a default license by Github which gives it a bit more power, and is unlikely to ever be contested.

> The only reason I license as MIT is because it's a license that everyone is familiar with -- a license which people feel comfortable contributing with. Many people don't feel that way with public domain code (regardless of "correctness" for their feelings).

I haven’t seen those feelings widely expressed. I haven’t done the research either, so I won’t contest that.

> Is there a practical problem you're facing with MIT which you could overcome with public domain code?

Yes. All of my [`alfred-workflows`](https://github.com/vitorgalvao/alfred-workflows) ([Alfred](https://www.alfredapp.com/) is an OS X app, “workflows” are kind of plugins for it) are public domain. Workflows are easy enough to start making that a non-programmer can do it, but can grow very powerful (in a way that requires extensive programming). In [their forums](http://www.alfredforum.com/) I encounter people benefiting from that regularly, who want to do more with Alfred and learn more programming. I remember when I started out licenses were always a bore. Even the MIT license isn’t very explicit: at which point have you modified the code so much that it’s no longer the original, and the license is no longer required?

When I make those workflows, I want everyone to be able to use and learn from them *without restriction*. Go nuts. This for a long time prevented me from including non-public-licensed code (`youtube-dl` could go in) and even made me build [`notificator`](https://github.com/vitorgalvao/notificator), which is a non-restricted alternative to [`terminal-notifier`](https://github.com/julienXX/terminal-notifier). The former is less capable than the later in most regards, but still it saw interest, which means other people care as well.

Currently I do include non-plublic-domain code, but I place it all under a `_licensed` directory on the root of the project and need to be vocal on explaining what code is what license, while still making people sure they can use most of the code however they want. Needing the caveat that they have to check first isn’t a good experience.

In short, having code encumbered by a license, even if its only restriction is “this license text file needs to be included”, is a big enough congnitive load and can mess up your code organization. You no longer do what’s best, but what accomodates the license.

@zacharyvoase Also made [a post on this](http://zacharyvoase.com/2010/01/04/unlicense/) a while back.

All that said, you have answered the question very well and made the change, so this can be closed. Thank you for a very polite and pleasant exchange. I’m going now to read in full the links you provided.

**Comment by cdown** [comment]
I just realised I typoed -- yturl was under an ISC license, not an MIT license.

I hope this outcome basically resolves your concerns about the public domain: https://github.com/cdown/yturl/blob/develop/LICENSE.rst

I have some reservations about Unlicense, but hopefully giving people the choice between ISC and Unlicense will allow them to do whatever they feel is right.
